src/Concerns/HasBotsAndChats.php

```diff
@@ -341,6 +341,19 @@ public function setChatPhoto(string $path): Telegraph
         return $telegraph;
     }
 
+    public function setMessageReaction(string $messageId, array $reaction, bool $isBig = false): Telegraph
```

Comment:
I believe it makes little sense to separate them, as Telegram accepts only 2 (3) types of emojis. We need to pass the type as either 'emoji' or 'custom_emoji' (or 'paid', which should not be allowed to bots). I didn’t want to use two separate parameters since they are dependent on each other. Creating 2 (3) methods to handle the same task also isn’t an ideal solution.

Reply:
Also, I would add a list of possible reactions to the constants class (enums are [not used](https://github.com/defstudio/telegraph/tree/main/src/Enums) in this project) because Telegram allows certain emoji to be sent as [reactions](https://core.telegram.org/bots/api#reactiontypeemoji).

Reply:
As I mentioned in my previous message, I didn’t want to hardcode the list of emojis in the library, in case Telegram updates it. Additionally, it's possible to send more emojis than those listed by using the 'custom_emoji' type.

Reply:
I agree. It gives more choice for the developer.

Reply:
> I think this could have a better dx
> 
> Something like reactWithEmoji($messageId, '👍', €$isBig)

@Anze  I was not meaning to hardcode all emojis, this is an example of what I want to achieve as a method call

i think that instead of have a method like this

```php
function setMessageReaction(string $messageId, array $reaction, bool $isBig = false) 
```

where you have to pass an array that specifies `type = emojy` and the actual emojy, we could make a syntactic sugar to avoid telling `type = emojy`:

```php
function reactWithEmoji(string $messageId, string $emojy, bool  $isBig = false)
```

Reply:
@fabio-ivona In that case, two more methods would be required:

```php
reactWithCustomEmoji(string $messageId, string $emojiId);
reactWithPaid();
```

- https://core.telegram.org/bots/api#reactiontypecustomemoji
- https://core.telegram.org/bots/api#reactiontypepaid - The method always takes the `type` property as the `paid` value, so you don't need to pass anything from the outside.

Reply:
yep, I agree

I think it would make a better DX

Reply:
The only thing to consider is the possibility of multiple method calls. For example:

```php
// ...
->reactWithEmoji(123, '👍', true)
->reactWithEmoji(123, '🥰', false)
->reactWithEmoji(123, '🔥', false)
->reactWithCustomEmoji(123, '222')
->reactWithCustomEmoji(123, '333')
->reactWithCustomEmoji(123, '444')
->reactWithPaid()
->send()
```
Because the [setMessageReaction](https://core.telegram.org/bots/api#setmessagereaction) method takes an array of emoji as the value for the `reaction` field.

Reply:
The bot can only react with a single emoji per message, even though the parameters suggest it supports a list. That's why it's `array $reaction` and not `array $reactionS`.

Also, custom emojis are only allowed if they're already present on the message or permitted by the admins. I assume it would be rare usage. IMHO it is easier to use as param and not another rare used method.

Sample, I prefer: 
```
if ($customReaction) {
  $reaction = ['type'=>'custom_emoji', 'emoji'=>'123213124123421411234'];
} else {
  $reaction = ['type'=>'emoji', 'emoji'=>'👍'];
}
$chat->reactWithEmoji($messageId, $reaction)->send();
```
instead of:
```
if ($customReaction) {
  $chat->reactWithCustomEmoji($messageId, '123213124123421411234')->send();
} else {
  $chat->reactWithEmoji($messageId, '👍')->send();
}
```

Paid emojis aren’t accessible to bots, so creating a method that would always fail doesn’t make sense.

In short, chaining methods for this method isn’t practical.

Reply:
@Anze I tested sending reactions in a single message and this is what I found out. Despite the obvious array transmission, Telegram allows only one emoji character to be sent, otherwise it returns a `400 REACTIONS_TOO_MANY` error.

And also noticed another behavior:

- **Person**. I can put multiple emoji under one post and all of them will be displayed. For example, if I put “👍” and after that I added “🔥”, my two reactions will be “👍, 🔥” under the post.
- **Bot**. In fact it can send only one reaction and it will overwrite the one sent earlier. If a bot sent a “👍” and then sent a “🔥”, then the “👍” reaction will be overwritten and Telegram will show only the “🔥” reaction.

In this regard, I agree that there can be only one method, and a paid method is not necessary at all :)

Reply:
> ```php
> if ($customReaction) {
>   $reaction = ['type'=>'custom_emoji', 'emoji'=>'123213124123421411234'];
> } else {
>   $reaction = ['type'=>'emoji', 'emoji'=>'👍'];
> }
> $chat->reactWithEmoji($messageId, $reaction)->send();
> ```

I would use a ternary operator instead. It would be prettier that way, in my opinion:

```php
$reaction = $customReaction
    ? ['type'=>'custom_emoji', 'emoji'=>'123213124123421411234']
    : ['type'=>'emoji', 'emoji'=>'👍'];

$chat->reactWithEmoji($messageId, $reaction)->send();
```
🙂

Reply:
> The bot can only react with a single emoji per message, even though the parameters suggest it supports a list. That's why it's `array $reaction` and not `array $reactionS`.
> 
> Also, custom emojis are only allowed if they're already present on the message or permitted by the admins. I assume it would be rare usage. IMHO it is easier to use as param and not another rare used method.
> 
> Sample, I prefer:
> 
> ```
> if ($customReaction) {
>   $reaction = ['type'=>'custom_emoji', 'emoji'=>'123213124123421411234'];
> } else {
>   $reaction = ['type'=>'emoji', 'emoji'=>'👍'];
> }
> $chat->reactWithEmoji($messageId, $reaction)->send();
> ```
> 
> instead of:
> 
> ```
> if ($customReaction) {
>   $chat->reactWithCustomEmoji($messageId, '123213124123421411234')->send();
> } else {
>   $chat->reactWithEmoji($messageId, '👍')->send();
> }
> ```
> 
> Paid emojis aren’t accessible to bots, so creating a method that would always fail doesn’t make sense.
> 
> In short, chaining methods for this method isn’t practical.

@Anze I think we can keep the `setMessageReaction()` method and adding the two `reactWithEmoj()` and `reactWithCustomEmoji()` methods as syntactic sugar to achieve both results

the aim of this package is to allow a fluent and easy interaction with telegam, so I feel this could be a better solution for both points of view